File: profile_service/src/main/java/huy/project/profile_service/core/domain/dto/response/PageInfo.java
package huy.project.profile_service.core.domain.dto.response;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PageInfo {
    private Long totalPage;
    private Long totalRecord;
    private Long pageSize;
    private Long nextPage;
    private Long previousPage;
}

File: location_service/LocationService.Core/Domain/Dto/Response/PageInfo.cs
namespace LocationService.Core.Domain.Dto.Response;

public class PageInfo
{
    public int totalPage { get; set; }
    public int totalRecord { get; set; }
    public int pageSize { get; set; }
    public int nextPage { get; set; }
    public int previousPage { get; set; }

    public static PageInfo ToPageInfo(int currentPage, int pageSize, int totalRecord)
    {
        int totalPage = totalRecord / pageSize;
        if (totalRecord % pageSize > 0)
        {
            totalPage++;
        }

        return new PageInfo
        {
            totalPage = totalPage,
            totalRecord = totalRecord,
            pageSize = pageSize,
            nextPage = currentPage < totalPage ? currentPage + 1 : totalPage,
            previousPage = currentPage > 0 ? currentPage - 1 : 0
        };
    }
}